masquerade-sample/src/main/java/com/flipkart/masquerade/test/TestConfig.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.masquerade.test;
+
+import com.flipkart.masquerade.Configuration;
+import com.flipkart.masquerade.rule.*;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by shrey.garg on 25/04/17.
+ */
+public class TestConfig implements Configuration {
+    private static Set<Rule> rules = new HashSet<>();
+
+    static {
+        Rule rule = new Rule(
+                "VaAn",
+                ValidationAnnotation.class,
+                Eval.class,
+                new CompositeRule(
+                        new BasicRule("name", Operator.EQUAL, "getPlatform()"),
+                        new CompositeRule(Conjunction.OR,
+                                new BasicRule("since", Operator.LESSER, "getVersion()"),
+                                new BasicRule("till", Operator.GREATER, "getVersion()")
+                        )
+                )
+        );
+        rules.add(rule);
```

Comment:
This is little convoluted to me. I just wanted to extend this sample to do cloaking based on a simple string field called client. Hence added a ClientAnnotation and a rule like below.

`
public @interface ClientAnnotation {
    String value();
}

@ClientAnnotation("A")
    private One one;

Rule rule1 = new Rule(
    "Client",
    ClientAnnotation.class,
    String.class,
    new CompositeRule(new BasicRule("value", Operator.UNEQUAL, "toString()"))
    );`

A couple of problems in this approach
1. By looking at ClientAnnotation usage, I can't categorically say "one" will be set to null if client == "A" or if client != "A". I need to go and look at the rule which is using this annotation and look at the CompositeRule inside it (in above case if its not A, it will set to null). I feel, this expression has to be expressed somehow in the annotation itself - not sure if its doable.
2. The object containing a reference to "One one" which is using a ClientAnnotation, if it has to use another validation annotation (say platform validation annotation) on another field or same field (one), then the generated cloak has two mask classes against the source object class. Obviously, the second put() in map will be picked up at runtime and hence one of the masks won't come into effect. Either we need to convert the map value to contain a list of mask objects or force to have only one validation covering all cases (the its not a set of rules, just a single rule always)
3. Readability of the Rule as well as type safety goes for a toss as we are referring method names of Eval object and fields of annotations using strings.